peas/weather_web_data.py

```diff
@@ -0,0 +1,188 @@
+    not safe.
+    Data is also given the weather condition, for example if the wind value is greater
+    that the one specified the weather condition will be 'windy'.
+
+    Once all the required data has been defined and given conditions, it will then
+    decide if the system is safe. If one value is False then the safety condition of
+    the system is False. All entries must be True so that the safety condition can
+    return True.
+
+    The final conditions and values are then sent to the dome controller to either
+    leave, open or close the dome. They are also saved in a database so that previous
+    entries can be retrived.
+    """
+
+    def __init__(self, use_mongo=True):
+        super.__init__(self, use_mongo=use_mongo)
+
+        self.logger = logging.getLogger(self.web_data.get('name'))
+        self.logger.setLevel(logging.INFO)
+
+        self.max_age = TimeDelta(self.web_data.get('max_age', 60.), format='sec')
+
+        self.table_data = None
+
+
+    def capture(self, use_mongo=False, send_message=False, **kwargs):
+        self.logger.debug('Updating weather data')
+
+        data = {}
+
+        data['weather_data_from'] = self.web_data.get('name')
+        self.table_data = self.fetch_met_data()
+        col_names = self.web_data.get('column_names')
+        for i in range(0, len('col_names')):
+            data[col_names[i]] = self.table_data[col_names[i]][0]
+
+        super.capture()
+
+        return data
+
+    def make_safety_decision(self, current_values):
+        self.logger.debug('Making safety decision with {}'.format(self.web_data.get('name')))
+        super.make_safety_decision(current_values)
+
+        return {'Safe': safe,
+                'Sky': cloud[0],
+                'Wind': wind[0],
+                'Gust': gust[0],
+                'Rain': rain[0]}
+
+    def fetch_met_data(self):
+        try:
+            cache_age = Time.now() - self._met_data['Time (UTC)'][0] * 86400
+        except AttributeError:
+            cache_age = self.web_data.get('cache_age', 1.382e10) * u.year
+
+        if cache_age > self.max_age:
+            # Download met data file
+            m = requests.get(self.web_data.get('link'))
+            # Remove the annoying " and newline between the date and time
+            met = m.text.replace('."\n',' ')
+            # Remove the " before the date
+            met = met.replace('" ', '')
+
+            # Parse the tab delimited met data into a Table
+            t = Table.read(met, format='ascii.no_header', delimiter='\t',
+                                names=self.web_data.get('column_names'))
+
+            # Convert time strings to Time
+            t['Time (UTC)'] = Time(t['Time (UTC)'], format='mixed_up_time')
+            # Change string format to ISO
+            t['Time (UTC)'].format = 'iso'
+            # Convert from AAT standard time to UTC
+            t['Time (UTC)'] = t['Time (UTC)'] - 10 * u.hour

```

Comment:
Once the code is running and working well, we can do that. Would it just be making a subclass that contains the code to change the time format and then calling the class into the current one?
Just trying to understand how it would work. :)

Reply:
Yes, something like that. A class that parses sensibly formatted tables of met data in a fairly general, adaptable way, and a subclass of that that adds the extra processing steps specific to the AAT (pruning surplus newlines & quotation marks, manual timezone conversion from AAT standard time to UTC).